cumulusci/cli/cci.py

```diff
@@ -807,15 +807,50 @@ def org_connect(config, org_name, sandbox, login_url, default, global_org):
         scope="web full refresh_token",
     )
     oauth_dict = oauth_capture()
+    # if sandbox:
+    #     org_config = ScratchOrgConfig(oauth_dict, org_name)
+    # else:
+    #     org_config = OrgConfig(oauth_dict, org_name)
+    #     org_config.load_userinfo()
+    #   unsure whether this logic is sound due to the fact that trialforce orgs are sandboxes
+    #   Need to validate this with someone. <-------------------***********NOTE***************
+    
     org_config = OrgConfig(oauth_dict, org_name)
     org_config.load_userinfo()
-
-    config.keychain.set_org(org_config, global_org)
-
+    if sandbox:
+        username = str(vars(org_config)["config"]["userinfo"]["preferred_username"])
+        result = sfdx(f"force:org:display -u {username}" )
+        config_value = (
+            io.TextIOWrapper(result.stdout, encoding=sys.stdout.encoding).read()
+        )
+        parsed = config_value.split("\n")
+        # skipping first two header rows
+        for i in range(3,len(parsed)-1):
+            value = parsed[i].split(' ')
+            if value[0].lower() == "expiration" and value[len(value)-1] != '':
+                org_config = ScratchOrgConfig(oauth_dict,org_name)
+                org_config.config.update({"username": username}) 
     if default:
         config.keychain.set_default_org(org_name)
         click.echo(f"{org_name} is now the default org")
 
+    org_config.refresh_oauth_token(config.keychain)
```

Comment:
What's this for? We should already have a fresh access token from the oauth flow above.

Reply:
Not sure why I have this if I'm being totally honest